docs/architecture/adr-070-unordered-account.md

```diff
@@ -2,214 +2,134 @@
-in sequential order. However it makes it tricky to send many transactions from the
-same sender concurrently in a reliable way. IBC relayer and crypto exchanges are
-typical examples of such use cases.
+Account sequence values serve to prevent replay-attacks and ensure transactions from the same sender are included into blocks and executed
+in sequential order. Unfortunately, this makes it difficult to reliably send many concurrent transactions from the
+same sender. Victims of such limitations include IBC relayers and crypto exchanges.
 
 ## Decision
 
-We propose to add a boolean field `unordered` to transaction body to mark "un-ordered"
-transactions.
+We propose adding a boolean field `unordered` and a uint64 field `timeout_timestamp` to the transaction body.
 
-Un-ordered transactions will bypass the nonce rules and follow the rules described
-below instead, in contrary, the default ordered transactions are not impacted by
-this proposal, they'll follow the nonce rules the same as before.
+Unordered transactions will bypass the traditional account sequence rules and follow the rules described
+below, without impacting traditional ordered transactions; they'll follow the sequence rules the same as before.
 
-When an un-ordered transaction is included into a block, the transaction hash is
-recorded in a dictionary. New transactions are checked against this dictionary for
-duplicates, and to prevent the dictionary grow indefinitely, the transaction must
-specify `timeout_timestamp` for expiration, so it's safe to removed it from the
-dictionary after it's expired.
+We will introduce new storage of time-based, ephemeral unordered sequences using the SDK's existing KV Store library.
 
-The dictionary can be simply implemented as an in-memory golang map, a preliminary
-analysis shows that the memory consumption won't be too big, for example `32M = 32 * 1024 * 1024`
-can support 1024 blocks where each block contains 1024 unordered transactions. For
-safety, we should limit the range of `timeout_timestamp` to prevent very long expiration,
-and limit the size of the dictionary.
+When an unordered transaction is included in a block, a concatenation of the `timeout_timestamp` and sender’s bech32 address
+will be recorded to state (i.e. `542939323/cosmos1v1234567890AbcDeF`). In cases of multi-party signing, we will use a
+comma-separated list of the addresses that signed the transaction (i.e. `5532231/cosmosv11,cosmosv12,cosmosv13`)
+
+New transactions will be checked against the state to prevent duplicate submissions. To prevent the state from growing indefinitely, we propose the following:
+
+- Define an upper bound for the value of `timeout_timestamp` (i.e. 10 minutes).
+- Extend the PreBlocker method to remove state entries with a `timeout_timestamp` earlier than the current block time.
 
 ### Transaction Format
 
 ```protobuf
 message TxBody {
   ...
-
+          
   bool unordered = 4;
+  google.protobuf.Timestamp timeout_timestamp = 5
```

Comment:
this was just missing from the prev spec right?

Reply:
yes